pallets/payment-streams/src/utils.rs

```diff
@@ -547,177 +550,253 @@ where
+                        },
+                    )?;
+
+                    // Update the total amount charged:
+                    total_amount_charged = total_amount_charged
                         .checked_add(&amount_to_charge)
-                        .is_some(),
-                    ArithmeticError::Overflow
-                );
-
-                // Charge the payment stream from the user's balance
-                T::NativeBalance::transfer(
-                    user_account,
-                    &provider_payment_account,
-                    amount_to_charge,
-                    Preservation::Preserve,
-                )?;
-
-                // Set the last charged block to the block number of the last chargeable block
-                FixedRatePaymentStreams::<T>::mutate(
-                    provider_id,
-                    user_account,
-                    |payment_stream| {
-                        let payment_stream = expect_or_err!(
-                            payment_stream,
-                            "Payment stream should exist if it was found before.",
-                            Error::<T>::PaymentStreamNotFound
-                        );
-                        payment_stream.last_charged_block = payment_stream.last_chargeable_block;
-                        Ok::<(), DispatchError>(())
-                    },
-                )?;
-
-                // Update the total amount charged:
-                total_amount_charged = total_amount_charged
-                    .checked_add(&amount_to_charge)
-                    .ok_or(Error::<T>::ChargeOverflow)?;
+                        .ok_or(Error::<T>::ChargeOverflow)?;
+                }
             }
         }
 
         // If the dynamic-rate payment stream exists:
         if let Some(dynamic_rate_payment_stream) = dynamic_rate_payment_stream {
             // Calculate the difference between the last charged price index and the price index at the last chargeable block
-            let price_index_difference = expect_or_err!(dynamic_rate_payment_stream
-				.price_index_at_last_chargeable_block
-				.checked_sub(&dynamic_rate_payment_stream.price_index_when_last_charged), "Last chargeable price index should always be greater than or equal to the last charged price index, inconsistency error.",
-				Error::<T>::LastChargedGreaterThanLastChargeable);
-
-            // Calculate the amount to charge
-            let amount_to_charge = price_index_difference
-                .checked_mul(&dynamic_rate_payment_stream.amount_provided.into())
-                .ok_or(Error::<T>::ChargeOverflow)?;
-
-            // Check the free balance of the user
-            let user_balance = T::NativeBalance::reducible_balance(
-                &user_account,
-                Preservation::Preserve,
-                Fortitude::Polite,
-            );
+            // Note: If the last chargeable price index is less than the last charged price index, we charge 0 to the user
```

Comment:
```suggestion
            // Note: If the last chargeable price index is less than the last charged price index, we charge 0 to the user, because that would be an impossible state.
```